src/ValispaceInit.m

```diff
@@ -1,36 +1,86 @@
-function ValispaceInit(URL,Username,Password,insecure)
+function ValispaceInit(URL, APIToken, insecure)
+    % ValispaceInit Initializes the connection to the Valispace API using a Bearer Token
+    %
+    % Parameters:
+    %   URL       - The base URL of the Valispace instance (e.g., 'https://demonstration.valispace.com')
+    %   APIToken  - Your Valispace API Access Token
+    %   insecure  - (Optional) If set to 'insecure', allows HTTP connections (not recommended)
 
-% check for SSL connection, before sending the username and password
-    if (strncmpi(URL,'http://',7))
-        if (exist('insecure', 'var') && strcmp(insecure, 'insecure'))
-            warning('You are sending your password-credentials using an unencrypted connection. Better use "https://..." instead!');
+    % Declare ValispaceLogin as global
+    global ValispaceLogin
+
+    %% Check Number of Inputs
+    narginchk(2, 3); % URL and APIToken are required, insecure is optional
+
+    %% Step 1: Validate Inputs
+    if ~(ischar(URL) || isstring(URL)) || ~isscalar(URL)
+        error('VALISPACE-ERROR: URL must be a scalar string or character vector.');
+    end
+    if ~(ischar(APIToken) || isstring(APIToken)) || ~isscalar(APIToken)
+        error('VALISPACE-ERROR: APIToken must be a scalar string or character vector.');
+    end
+
+    %% Step 2: Check for SSL Connection
+    if startsWith(URL, 'http://', 'IgnoreCase', true)
+        if nargin == 3 && strcmpi(insecure, 'insecure')
+            warning(['You are sending your credentials over an unencrypted connection. ' ...
+                     'It is highly recommended to use "https://..." instead!']);
         else
-            error('VALISPACE-ERROR: You are trying to send your password-credentials using an unencrypted connection. Better use "https://..." instead! If you still want to go ahead with an insecure connection, please use ValisapceInit() with "insecure" as a last argument.');
+            error(['VALISPACE-ERROR: You are attempting to send credentials over an unencrypted ' ...
+                   'connection. Please use "https://..." for secure communication. ' ...
+                   'If you intend to proceed with an insecure connection, use ValispaceInit() ' ...
+                   'with "insecure" as the last argument.']);
         end
     end
```

Comment:
@joao-goncalves-altium should we allow insecure connections to be established with our integrations??